### PR TITLE
Fix --use-clang-cl to actually use clang-cl

### PR DESCRIPTION
**Pull request by agrieve**
A few fixes:
 * Fixed the flag
 * Fixed one compile error that surfaces with clang-cl
 * Fix js2c.py's conversion of .gypi -> .json
 * Fixed a depot_tools issue unrelated to this change, but that I realized was incorrect in the previous pull request.

### CODE REVIEW

File: src/node_win32_perfctr_provider.h [hunk 1]
```diff
@@ -32,7 +32,7 @@
 
 namespace node {
 
-extern HANDLE NodeCounterProvider;
+EXTERN_C HANDLE NodeCounterProvider;
```
Comment (bnoordhuis) [inline src/node_win32_perfctr_provider.h]: This doesn't look Obviously Correct to me.  I would expect it to result in duplicate definitions when the header file is included in more than a single source file.

**Comment by agrieve** [comment]
This was certainly one of those change-it-until-the-compile-error-goes-away
kind of fixes, so I'm happy to know what the right fix here is. What clang
was complaining of was that NodeCounterProvider was being defined with a
different linkage than it was declared having. The linker then complained
that it was being redefined (once in this file, and once in a generated
file of the same name).

**Comment by bnoordhuis** [comment]
That's probably because it's `EXTERN_C DECLSPEC_SELECTANY` in the .cc file but not in the .h.

What clang version is it?  I don't think clang < 5.0.1 knows about `__declspec(selectany)` (what `DECLSPEC_SELECTANY` is a conditional definition of.)

**Comment by hashseed** [comment]
I merged the part without changes to `src/node_win32_perfctr_provider.*` for now.

**Pull request merged**